tools/pngkit.py

```
@@ -0,0 +1,211 @@
+    return b if pb <= pc else c
+
+
+_CHANNELS = {0: 1, 2: 3, 3: 1, 4: 2, 6: 4}  # grayscale, RGB, indexed, gray+alpha, RGBA
+
+
+def _unfilter(raw, h, stride, bpp):
+    """Reverse PNG scanline filters, returning the concatenated unfiltered scanlines."""
+    out, prev = bytearray(), bytearray(stride)
+    pos = 0
+    for _ in range(h):
+        filt = raw[pos]; pos += 1
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)